controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -151,7 +151,12 @@ func NewClient(ctx context.Context, config ClientConfiguration) (*Client, error)
 		return nil, errors.Wrap(err, "unable to create etcd client")
 	}
 
-	return newEtcdClient(ctx, etcdClient)
+	client, err := newEtcdClient(ctx, etcdClient)
+	if err != nil {
+		closeErr := etcdClient.Close()
+		return nil, errors.Wrap(err, closeErr.Error())
```

Comment:
```suggestion
		return nil, errors.Wrap(kerrors.NewAggregate([]error{err, closeErr}), "unable to create etcd client")
```

Reply:
Let's aggregate the errors like this. NewAggregate will also filter out closeErr if it's nil